src/boot/bootctl-install.c

```diff
@@ -753,31 +980,61 @@ static int are_we_installed(const char *esp_path) {
 
         /* Tests whether systemd-boot is installed. It's not obvious what to use as check here: we could
          * check EFI variables, we could check what binary /EFI/BOOT/BOOT*.EFI points to, or whether the
-         * loader entries directory exists. Here we opted to check whether /EFI/systemd/ is non-empty, which
-         * should be a suitable and very minimal check for a number of reasons:
-         *
-         *  → The check is architecture independent (i.e. we check if any systemd-boot loader is installed,
-         *    not a specific one.)
-         *
-         *  → It doesn't assume we are the only boot loader (i.e doesn't check if we own the main
-         *    /EFI/BOOT/BOOT*.EFI fallback binary.
-         *
-         *  → It specifically checks for systemd-boot, not for other boot loaders (which a check for
-         *    /boot/loader/entries would do). */
-
-        _cleanup_free_ char *p = path_join(esp_path, "/EFI/systemd/");
-        if (!p)
+         * loader entries directory exists. Checking for /EFI/systemd/systemd-boot*.efi is no longer enough
+         * as we now support shim-based installations, where the bootloader will be in a distribution
+         * specific directory. So first check for /EFI/systemd/systemd-boot*.efi, if that is not found then
+         * check whether /loader/entries exists and is not empty, if that is not the case then check if
+         * /loader/loader.conf exists, and if not finally check if /loader/entries.srel exists. */
```

Comment:
i dont follow here. why make this change?

the exsting check explicitly didn't bother with checking what the entrypoint was pointing to, but just checks where *our* resources are placed, ignoring where the entrypoint is pointing to. Why would we suddenly care about the entrypoint?